src/common/entity/domain_icon.ts

```diff
@@ -72,6 +77,26 @@ export const domainIcon = (
     case "media_player":
       return compareState === "playing" ? mdiCastConnected : mdiCast;
 
+    case "switch":
+      switch (stateObj?.attributes.device_class) {
+        case "outlet":
+          switch (state) {
+            case "on":
+              return mdiPowerPlug;
+            default:
+              return mdiPowerPlugOff;
+          }
```

Comment:
But don't use a switch here
```suggestion
          return state === "on" ? mdiPowerPlug : mdiPowerPlugOff;
```